src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ActiveEditorContextTracker.cs

```diff
@@ -25,21 +25,17 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
     [AppliesTo(ProjectCapability.DotNetLanguageService)]
     internal class ActiveEditorContextTracker : IActiveIntellisenseProjectProvider, IVsContainedLanguageProjectNameProvider, IActiveEditorContextTracker
     {
-        private readonly IActiveWorkspaceProjectContextHost _activeWorkspaceProjectContextHost;
-        private readonly IProjectThreadingService _threadingService;
-        private ImmutableDictionary<IWorkspaceProjectContext, string> _contexts = ImmutableDictionary<IWorkspaceProjectContext, string>.Empty;
+        private ImmutableList<string> _contexts = ImmutableList<string>.Empty;
         private string _activeIntellisenseProjectContext;
 
         [ImportingConstructor]
-        public ActiveEditorContextTracker(IProjectThreadingService threadingService, IActiveWorkspaceProjectContextHost activeWorkspaceProjectContextHost)
+        public ActiveEditorContextTracker()
         {
-            _threadingService = threadingService;
-            _activeWorkspaceProjectContextHost = activeWorkspaceProjectContextHost;
         }
 
         public string ActiveIntellisenseProjectContext
         {
-            get { return _activeIntellisenseProjectContext ?? GetWorkspaceContextIdFromActiveContext(); }
+            get { return _activeIntellisenseProjectContext ?? _contexts.FirstOrDefault(); }
```

Comment:
Might not make a difference or be worth changing, but `FirstOrDefault` could have just been used on the `Values` property of the ImmutableDictionary

Reply:
ImmutableDictionary doesn't have order semantics, meaning that it would no longer be the first one registered.

Reply:
Makes sense, thanks